From b4a6c562cde4e064494e261bc3f64000cbf628a5 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Thu, 14 Jan 2021 15:23:39 +0100
Subject: [PATCH] Don't manually call reverseTwinsChanged when destructing
 FormLayout

This line causes a crash when destructing the screenlocker KCM.
It doesn't seem to be caused by reverseTwinsChanged itself but rather by
accessing the Qt object. I don't fully undestand the root cause.

However the call itself appears to be unnecessary since in the line
above we write to reverseTwins and that triggers the changed signal if
needed.

BUG: 431419
BUG: 428461
---
 src/controls/FormLayout.qml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/controls/FormLayout.qml b/src/controls/FormLayout.qml
index 74292fda..a27903f2 100644
--- a/src/controls/FormLayout.qml
+++ b/src/controls/FormLayout.qml
@@ -84,7 +84,6 @@ Item {
     Component.onDestruction: {
         for (let i in twinFormLayouts) {
             twinFormLayouts[i].children[0].reverseTwins = twinFormLayouts[i].children[0].reverseTwins.filter(function(value, index, arr){ return value != root;})
-            Qt.callLater(() => twinFormLayouts[i].children[0].reverseTwinsChanged());
         }
     }
     GridLayout {
-- 
GitLab

